File: 2308-divide-array-into-equal-pairs/2308-divide-array-into-equal-pairs.ts
function divideArray(nums: number[]): boolean {
    const mp = new Map<number, number>();

    for (let num of nums) {
        mp.set(num, (mp.get(num) || 0) + 1);
    }

    return [...mp.values()].every((num) => num % 2 === 0);
};

File: 2308-divide-array-into-equal-pairs/2308-divide-array-into-equal-pairs.java
class Solution {
    public boolean divideArray(int[] nums) {
        Map<Integer, Integer> mp = new HashMap<>();

        for (int num : nums) {
            mp.put(num, mp.getOrDefault(num, 0) + 1);
        }

        for (int num : mp.values()) {
            if (num % 2 != 0)
                return false;
        }

        return true;
    }
}